exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporter.java

```diff
@@ -0,0 +1,198 @@
+  @VisibleForTesting static final String STATUS_CODE = "otel.status_code";
+  @VisibleForTesting static final String STATUS_DESCRIPTION = "otel.status_description";
+  @VisibleForTesting static final String STATUS_ERROR = "error";
+
+  private final SpanBytesEncoder encoder;
+  private final Sender sender;
+  private final Endpoint localEndpoint;
+
+  ZipkinSpanExporter(SpanBytesEncoder encoder, Sender sender, String serviceName) {
+    this.encoder = encoder;
+    this.sender = sender;
+    this.localEndpoint = produceLocalEndpoint(serviceName);
+  }
+
+  /** Logic borrowed from brave.internal.Platform.produceLocalEndpoint */
+  static Endpoint produceLocalEndpoint(String serviceName) {
+    Endpoint.Builder builder = Endpoint.newBuilder().serviceName(serviceName);
+    try {
+      Enumeration<NetworkInterface> nics = NetworkInterface.getNetworkInterfaces();
+      if (nics == null) {
+        return builder.build();
+      }
+      while (nics.hasMoreElements()) {
+        NetworkInterface nic = nics.nextElement();
+        Enumeration<InetAddress> addresses = nic.getInetAddresses();
+        while (addresses.hasMoreElements()) {
+          InetAddress address = addresses.nextElement();
+          if (address.isSiteLocalAddress()) {
+            builder.ip(address);
+            break;
+          }
+        }
+      }
+    } catch (Exception e) {
+      // don't crash the caller if there was a problem reading nics.
+      if (logger.isLoggable(Level.FINE)) {
+        logger.log(Level.FINE, "error reading nics", e);
+      }
+    }
+    return builder.build();
+  }
+
+  static Span generateSpan(SpanData spanData, Endpoint localEndpoint) {
+    long startTimestamp = toEpochMicros(spanData.getStartEpochNanos());
+
+    long endTimestamp = toEpochMicros(spanData.getEndEpochNanos());
+
+    Span.Builder spanBuilder =
+        Span.newBuilder()
+            .traceId(spanData.getTraceId().toLowerBase16())
+            .id(spanData.getSpanId().toLowerBase16())
+            .kind(toSpanKind(spanData))
+            .name(spanData.getName())
+            .timestamp(toEpochMicros(spanData.getStartEpochNanos()))
+            .duration(endTimestamp - startTimestamp)
+            .localEndpoint(localEndpoint);
+
+    if (spanData.getParentSpanId().isValid()) {
+      spanBuilder.parentId(spanData.getParentSpanId().toLowerBase16());
+    }
+
+    for (Map.Entry<String, AttributeValue> label : spanData.getAttributes().entrySet()) {
+      spanBuilder.putTag(label.getKey(), attributeValueToString(label.getValue()));
+    }
+    Status status = spanData.getStatus();
+    if (status != null) {
+      spanBuilder.putTag(STATUS_CODE, status.getCanonicalCode().toString());
+      if (status.getDescription() != null) {
+        spanBuilder.putTag(STATUS_DESCRIPTION, status.getDescription());
+      }
+      if (!status.isOk()) {
+        spanBuilder.putTag(STATUS_ERROR, status.getCanonicalCode().toString());
```

Comment:
UI and dependency linking checks for errors. This is a simplified change, but check first of there's already a tag named "error"?

```suggestion
        spanBuilder.putTag("error", status.getCanonicalCode().toString());
```

Reply:
I'm not sure I understand what you mean. Are you suggesting that if the user had already added an "error" tag, that we don't try to overwrite it?

Reply:
yep that was what I meant

Reply:
gotcha

Reply:
this is now implemented.